### ISSUE TITLE
[BUG] tag are not reported with prediction when --multilabel_oaa is used

### ISSUE THREAD

**Issue by pommedeterresautee** [JohnLangford/vowpal_wabbit#1094]
Below is a simple example:

```
:~/test$ echo -e "1 | a\n2 | a b\n3 | a c\n2 | a b\n3 | b c\n1 | a c\n2 | d" | vw --multilabel_oaa 4 -f multilabel.model -p multilabel.train.out
final_regressor = multilabel.model
predictions = multilabel.train.out
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0        1                 2
1.000000 1.000000            2            2.0        2        1        3
1.000000 1.000000            4            4.0        2                 3

finished run
number of examples per pass = 7
passes used = 1
weighted example sum = 7.000000
weighted label sum = 0.000000
average loss = 1.142857
total feature number = 19
```

Then I run a prediction:

```
:~/test$ echo -e "'label1| a b c d\nlabel2| b d\nlabel3| ac" | vw -t -i multilabel.model -p ./predictions.txt --quiet
```

When I check the content of prediction.txt:

```

2

```

Please, remind that with `multilabel_oaa` it is possible to predict no label at all, so it explains why the first and the third line are empty. The issue is that there is no tag at all, but I asked for them (https://github.com/JohnLangford/vowpal_wabbit/wiki/Input-format).

I also tried:

```
:~/test$ echo -e "''label1 | a b c d\n'label2 | b d\n'label3 | ac" | vw -t -i multilabel.model -p ./predictions.txt --quiet
```

Unfortunately the result is the same.

Any idea how I can do to resolve that point?
I tried with last version of Vowpal (today), and compiled version 8.2.0 and 8.1.1 (compiled from Ubuntu).

Reading https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/multilabel_oaa.cc seems to me the tag is completly ignored :-(

The issue is VERY annoying because this command can't be used with `--daemon` because of this bug (no way to link a prediction to a request when requests are done in parallel).

Another point, which is not a bug but more a consistency issue, is that `multilabel_oaa` requires label from 0 to n-1, and any other ooa command requires labels from 1 to n. Seems `multilabel_oaa` is not a lot documented anyway.
https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/multilabel_oaa.cc#L41

**Comment by JohnLangford** [comment]
This is fixed in the current master.

**Comment by Sandy4321** [comment]
may you share some full example from 0 to end for 
--multilabel_oaa option

there is no --multilabel_oaa mentioning in 
https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Command-Line-Arguments


<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 14">
<meta name=Originator content="Microsoft Word 14">
<link rel=File-List
href="file:///C:\Users\sndr\AppData\Local\Temp\msohtmlclip1\01\clip_filelist.xml">

<link rel=themeData
href="file:///C:\Users\sndr\AppData\Local\Temp\msohtmlclip1\01\clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:\Users\sndr\AppData\Local\Temp\msohtmlclip1\01\clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-CA link=blue vlink=purple style='tab-interval:.5in'>


<p class=MsoNormal><a
href="https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/RunTests">https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/RunTests</a><o:p></o:p></p>



# Test 85: check multilabel_oaa
--
  | {VW} --multilabel_oaa 10 -d train-sets/multilabel   -p multilabel.predict




</body>

</html>
